### PR TITLE
Alexsherstinsky/documentation for 0.13 develop 11 30 2020 78

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* More expressive Datasource.self_check mechanism (utilizing adding a property to ExecutionEngine classes that reports on the parameters with which the given ExecutionEngine object was instantiated).
* Tests for the different Pandas and Spark use cases of the above.
* Tests for all utilities involved in the above.
* Fix of a typo in documentation for adding a self-managed Spark datasource.
* Fix to PandasExecutionEngine in the way it accepts `boto3_options`.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [ENHANCEMENT], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
-
-
-

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

Thank you for submitting!

### CODE REVIEW

File: great_expectations/util.py [hunk 1]
```diff
@@ -689,6 +752,89 @@ def lint_code(code):
...
+    return True
+
+
+def is_float(value: Any) -> bool:
+    try:
+        num: float = float(value)
+    except ValueError:
+        return False
+    return True
+
```
Comment (Shinnnyshinshin) [inline great_expectations/util.py]: Thank you for these. I may have written my own version of these in either the `sorters.py` or `partition_query.py` files, and I agree that `util.py` is the right place for them to live.

I'll adjust the other code to use these methods, and move/consolidate any repeating methods.

Comment (alexsherstinsky) [reply great_expectations/util.py]: @Shinnnyshinshin If possible, let us wait for this PR to be merged.  Thanks!

File: great_expectations/util.py [hunk 2]
```diff
@@ -61,6 +74,56 @@ def get_project_distribution() -> Union[Distribution, None]:
...
+    if not include_caller_names:
+        if call_args.get("cls"):
+            call_args_dict.pop("cls", None)
+        if call_args.get("self"):
+            call_args_dict.pop("self", None)
+
+    call_args_dict.update(**kwargs)
+
+    return call_args_dict
```
Comment (Shinnnyshinshin) [inline great_expectations/util.py:124]: this function looks great, but is honestly beyond my scope of knowledge. if you have time, could you walk me through some of the major steps? I'd love to learn more

**Review by Shinnnyshinshin — CHANGES_REQUESTED** [review]

File: great_expectations/util.py (hunk 1 above)
Comment (Shinnnyshinshin) [reply great_expectations/util.py]: absolutely

File: great_expectations/util.py (hunk 2 above)
Comment (Shinnnyshinshin) [reply great_expectations/util.py:124]: Thanks for the explanation :) resolved the question for me

**Review by Shinnnyshinshin — APPROVED** [review]
Looks wonderful. Thanks!

File: docs/guides/how_to_guides/configuring_datasources/how_to_configure_a_self_managed_spark_datasource.rst [hunk 3]
```diff
@@ -220,8 +220,9 @@ Additional Notes
                 class_name: Datasource
                 execution_engine:
                   class_name: SparkDFExecutionEngine
-                  spark_context:
-                    spark.master: local[*]
+                  spark_config:
```
Comment (abegong) [inline docs/guides/how_to_guides/configuring_datasources/how_to_configure_a_self_managed_spark_datasource.rst:223]: I don't understand this difference---trusting that you've already done UAT.

**Review by abegong — APPROVED** [review]
This is great. Thank you!

File: docs/guides/how_to_guides/configuring_datasources/how_to_configure_a_self_managed_spark_datasource.rst (hunk 3 above)
Comment (alexsherstinsky) [reply docs/guides/how_to_guides/configuring_datasources/how_to_configure_a_self_managed_spark_datasource.rst:223]: @abegong Previous version had insufficient indent and a missing property (i.e., we need the "spark_config" section).  I corrected it (and the tests confirm this).

**Comment by alexsherstinsky** [comment]
Updating to include the changes in the upstream (develop branch) and then will merge.  Thanks!

**Pull request merged**